modules/alarm-baseline/variables.tf

```diff
@@ -87,6 +87,11 @@ variable "cloudtrail_log_group_name" {
   description = "The name of the CloudWatch Logs group to which CloudTrail events are delivered."
 }
 
+variable "sns_topic_kms_master_key_id" {
+  description = "To enable SNS Topic encryption enter value with the ID or an alias (eg. alias/aws/sns) of a custom master KMS key that is used for encryption"
+  default     = null
```

Comment:
It seems like we need a customer managed CMK:

> If the default AWS Key Management Service (KMS) key "alias/aws/sns" is used for this encryption, then CloudWatch alarms can't publish messages to the SNS topic. The key policy of the default AWS KMS key for SNS doesn't allow CloudWatch alarms to perform "kms:Decrypt" and "kms:GenerateDataKey" API calls. Because this key is AWS managed, you can't manually edit the policy.

https://aws.amazon.com/premiumsupport/knowledge-center/cloudwatch-receive-sns-for-alarm-trigger/

Reply:
This is the same for S3 Events.

What if we implement #163. Move the CloudTrail CMK up a level, then use for both CloudTrail and SNS, but allow override.

Reply:
I did some testing `null` does work as of v0.12 (https://www.hashicorp.com/blog/terraform-0-12-conditional-operator-improvements)

I think we should merge this in and provide an example kms with policy in the docs and revisit including a default kms key for sns at a later time.